core/controllers/reader.py

```diff
@@ -629,6 +657,47 @@ class ExplorationCompleteEventHandler(base.BaseHandler):
     The state name recorded should be a state with a terminal interaction.
     """
 
+    URL_PATH_ARGS_SCHEMAS = {
+        'exploration_id': {
+            'schema': editor.SCHEMA_FOR_EXPLORATION_ID
+        }
+    }
+    HANDLER_ARGS_SCHEMAS = {
+        'POST': {
+            'collection_id': {
+                'schema': {
+                    'type': 'basestring'
+                },
+                'default_value': None
+            },
+            'version': {
+                'schema': editor.SCHEMA_FOR_VERSION
+            },
+            'state_name': {
+                'schema': {
+                    'type': schema_utils.SCHEMA_TYPE_BASESTRING
+                }
+            },
+            'session_id': {
+                'schema': {
+                    'type': schema_utils.SCHEMA_TYPE_BASESTRING
+                }
+            },
+            'client_time_spent_in_secs': {
+                'schema': {
+                    'type': schema_utils.SCHEMA_TYPE_FLOAT
+                }
+            },
+            'params': {
+                'schema': {
+                    'type': 'object_dict',
+                    'validation_method': (
+                        domain_objects_validator.validate_params),
+                }
+            }
```

Comment:
Why are the params send like this?

Reply:
Can some more structured manner be used?

Reply:
I think here I picked the wrong type. Later I found `params` arg in the domain layer and I have updated the schema accordingly. Please have a look and let me know if any changes are needed.